src/spetlr/power_bi/SparkPandasDataFrame.py

```diff
@@ -0,0 +1,314 @@
+from datetime import datetime
+from typing import Callable, Dict, List, Tuple, Union
+
+import numpy as np
+import pandas as pd
+from dateutil.parser import parse
+from pandas.core.generic import NDFrameT
+from pyspark.sql import DataFrame
+from pytz import timezone, utc
+
+from spetlr.exceptions import SpetlrException
+from spetlr.spark import Spark
+
+
+class SparkPandasDataFrame:
+    # converts data types from Spark to Pandas and vice versa
+    SPARK_TO_PANDAS_TYPES = {
+        "long": "Int64",
+        "int": "Int64",
+        "timestamp": "datetime64[ns]",
+        "boolean": "bool",
+        "string": "object",
+        "default": "object",
+    }
+
+    def __init__(
+        self,
+        json: Union[Dict, List],
+        schema: List[Tuple[Union[str, Callable[[pd.DataFrame], NDFrameT]], str, str]],
+        *,
```

Comment:
Just an idea:
I dont know why if it is deemed neccesary to explicitly define all other variables than json and schema - consider putting the asterisk in the beginning. Then the user must state expclitily also the json and the schema.

Reply:
These two parameters need always be specified. They are not optional. I like specifying parameters by name, but not necessarily with the required parameters. I think there are so many methods like this in all Python, so I would rather keep it this way. :) But thanks for suggesting.